### PR TITLE
chore: remove annotation layer FAB CRUD model view

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Remove FAB CRUD model view annotation layer, these endpoints are deprecated on FAB.

Renames routes from `/annotationlayermodelview/*` to `/annotationlayer/*`

Previous non REST API routes:
```
 <Rule '/annotationlayermodelview/api/read' (OPTIONS, HEAD, GET) -> AnnotationLayerModelView.api_read>,
 <Rule '/annotationlayermodelview/action_post' (OPTIONS, POST) -> AnnotationLayerModelView.action_post>,
 <Rule '/annotationlayermodelview/list/' (OPTIONS, HEAD, GET) -> AnnotationLayerModelView.list>,
 <Rule '/annotationlayermodelview/add' (OPTIONS, HEAD, POST, GET) -> AnnotationLayerModelView.add>,
 <Rule '/annotationmodelview/action_post' (OPTIONS, POST) -> AnnotationModelView.action_post>,
 <Rule '/annotationmodelview/list/' (OPTIONS, HEAD, GET) -> AnnotationModelView.list>,
 <Rule '/annotationmodelview/add' (OPTIONS, HEAD, POST, GET) -> AnnotationModelView.add>,
 <Rule '/annotationlayermodelview/delete/<pk>' (OPTIONS, HEAD, POST, GET) -> AnnotationLayerModelView.delete>,
 <Rule '/annotationlayermodelview/edit/<pk>' (OPTIONS, HEAD, POST, GET) -> AnnotationLayerModelView.edit>,
 <Rule '/annotationlayermodelview/show/<pk>' (OPTIONS, HEAD, GET) -> AnnotationLayerModelView.show>,
 <Rule '/annotationmodelview/delete/<pk>' (OPTIONS, HEAD, POST, GET) -> AnnotationModelView.delete>,
 <Rule '/annotationmodelview/edit/<pk>' (OPTIONS, HEAD, POST, GET) -> AnnotationModelView.edit>,
 <Rule '/annotationmodelview/show/<pk>' (OPTIONS, HEAD, GET) -> AnnotationModelView.show>,
 <Rule '/annotationmodelview/<pk>/annotation/' (OPTIONS, HEAD, GET) -> AnnotationModelView.annotation>,
```

Current non REST API routes:
```
<Rule '/annotationlayer/list/' (HEAD, OPTIONS, GET) -> AnnotationLayerView.list>
<Rule '/annotationlayer/<pk>/annotation' (HEAD, OPTIONS, GET) -> AnnotationLayerView.get>
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [x] Removes existing feature or API

### CODE REVIEW

File: superset/views/annotations.py [hunk 1]
```diff
@@ -14,117 +14,27 @@
...
-        if not form["start_dttm"].data and not form["end_dttm"].data:
-            raise StopValidation(_("annotation start time or end time is required."))
-        if (
-            form["end_dttm"].data
-            and form["start_dttm"].data
-            and form["end_dttm"].data < form["start_dttm"].data
-        ):
-            raise StopValidation(
-                _("Annotation end time must be no earlier than start time.")
-            )
```
Comment (villebro) [inline superset/views/annotations.py:47]: Note: I verified that the CRUD modal takes care of this validation in the frontend and that this validation is done in the command.

Bycatch while checking the related code: This is trying to append the `Type` to `exceptions`, should be an instance of `AnnotationDatesValidationError`

https://github.com/apache/superset/blob/ac0ff78616119bd5d8bebbb2781e0ef3486c4d19/superset/annotation_layers/annotations/commands/create.py#L73

Why on earth is MyPy not picking this up?

**Review by villebro — APPROVED** [review]
🔥 Code looks good and tested by creating a new set of annotations via the CRUD view and adding it to a chart via Explore (both event and interval). One bycatch nit that could be nice to get fixed while we're at it, other than that LGTM.

File: superset/views/annotations.py (hunk 1 above)
Comment (dpgaspar) [reply superset/views/annotations.py:47]: very nice catch! fixing

**Review by geido — APPROVED** [review]
LGTM

**Pull request merged**